pdr_backend/pdr_dashboard/callbacks/callbacks_home.py

```diff
@@ -224,3 +236,62 @@ def select_or_clear_all_predictoors(_, __, rows):
 
         ctx = dash.callback_context
         return select_or_clear_all_by_table(ctx, "predictoors_table", rows)
+
+    @app.callback(
+        [
+            Output("predictoor_config_modal", "is_open"),
+            Output("predictoor_addrs", "value"),
+        ],
+        [
+            Input("configure_predictoors", "n_clicks"),
+            Input("predictoor-addrs-local-store", "data"),
+        ],
+        prevent_initial_call=True,
+    )
+    def open_predictoors_config_modal(n_clicks, predictoor_addrs):
+        """
+        Select or clear all rows in the feeds table.
+        """
+        if predictoor_addrs and len(predictoor_addrs) > 0:
+            predictoor_addrs = "\n".join(predictoor_addrs)
+        else:
+            predictoor_addrs = ""
+        if n_clicks <= 0:
+            return [dash.no_update, predictoor_addrs]
+
+        return [True, predictoor_addrs]
+
+    @app.callback(
+        Output("predictoor-addrs-local-store", "data"),
+        Input("save_predictoors", "n_clicks"),
+        State("predictoor_addrs", "value"),
+    )
+    def save_predictoors_to_browser_storage(n_clicks, value):
+        if not value and not n_clicks:
+            return dash.no_update
+        addresses = value.split("\n")
+        addresses = [
+            addr.strip() for addr in addresses if addr.strip()
+        ]  # Remove empty lines
```

Comment:
addresses = filter(len, str_list) would probably have the same effect of removing empty lines

Reply:
took and looks and filter() filters out empty string, while strip() removes whitespaces which returns different results